components/Buttons/src/MDCFloatingButton.m

```diff
@@ -273,19 +341,168 @@ - (void)layoutSubviews {
   self.titleLabel.frame = UIEdgeInsetsInsetRect(self.titleLabel.frame, self.titleEdgeInsets);
 }
 
-#pragma mark - Subclassing
+#pragma mark - Property Setters/Getters
+
+- (void)setMode:(MDCFloatingButtonMode)mode {
+  BOOL needsShapeUpdate = _mode != mode;
+  _mode = mode;
+  if (needsShapeUpdate) {
+    [self updateShape:YES];
+  }
+}
+
+- (void)setMinimumSize:(CGSize)size
+              forShape:(MDCFloatingButtonShape)shape
+                inMode:(MDCFloatingButtonMode)mode {
+  NSMutableDictionary *modeToMinimumSize = self.shapeToModeToMinimumSize[@(shape)];
+  if (!modeToMinimumSize) {
+    modeToMinimumSize = [@{} mutableCopy];
+    self.shapeToModeToMinimumSize[@(shape)] = modeToMinimumSize;
+  }
+  modeToMinimumSize[@(mode)] = [NSValue valueWithCGSize:size];
+  if (shape == _shape && mode == _mode) {
+    [self updateShape:YES];
+  }
+}
+
+- (CGSize)minimumSizeForMode:(MDCFloatingButtonMode)mode {
+  NSMutableDictionary *modeToMinimumSize = self.shapeToModeToMinimumSize[@(_shape)];
+  if(!modeToMinimumSize) {
+    return CGSizeZero;
+  }
 
-- (UIEdgeInsets)defaultContentEdgeInsets {
-  return UIEdgeInsetsZero;
+  NSValue *sizeValue = modeToMinimumSize[@(mode)];
+  if (sizeValue) {
+    return [sizeValue CGSizeValue];
+  } else {
+    return CGSizeZero;
+  }
 }
 
-- (UIEdgeInsets)defaultHitAreaInsets {
-  switch (_shape) {
-    case MDCFloatingButtonShapeDefault:
-      return UIEdgeInsetsZero;
-    case MDCFloatingButtonShapeMini:
-      // Increase the touch target from (40, 40) to the minimum (48, 48)
-      return UIEdgeInsetsMake(-4, -4, -4, -4);
+- (BOOL)updateMinimumSize {
+  CGSize newSize = [self minimumSizeForMode:self.mode];
+  BOOL sizeChanged = (newSize.width > self.minimumSize.width)
+      || (newSize.height > self.minimumSize.height);
+  super.minimumSize = newSize;
+  if (sizeChanged) {
+    [self invalidateIntrinsicContentSize];
+  }
+  return sizeChanged;
+}
+
+- (void)setMaximumSize:(CGSize)size
+              forShape:(MDCFloatingButtonShape)shape
+                inMode:(MDCFloatingButtonMode)mode {
+  NSMutableDictionary *modeToMaximumSize = self.shapeToModeToMaximumSize[@(shape)];
+  if (!modeToMaximumSize) {
+    modeToMaximumSize = [@{} mutableCopy];
+    self.shapeToModeToMaximumSize[@(shape)] = modeToMaximumSize;
+  }
+  modeToMaximumSize[@(mode)] = [NSValue valueWithCGSize:size];
+  if (shape == _shape && mode == self.mode) {
```

Comment:
Nit: You use both self.mode and _mode across the code, maybe we should stick to one?

Reply:
`_mode` is only ever used to initialize the property. Did you mean `self.mode` and `_shape`

Reply:
Perhaps either always use _ivar or always use self.ivar. (Excepting init methods)
Mixed use in the same statement is less than ideal.

Reply:
Done